deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -0,0 +1,936 @@
+	}
+
+	// Deploy the prerequisite contracts to the new chain
+	err = runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+		return changeset.DeployPrerequisitesChangeset(e, c.prerequisiteConfigForNewChain())
+	}, newAddresses, e.ExistingAddresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run DeployPrerequisitesChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
+
+	// Deploy MCMS contracts
+	if c.MCMSDeploymentConfig != nil {
+		err = runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+			return commoncs.DeployMCMSWithTimelockV2(e, map[uint64]commontypes.MCMSWithTimelockConfigV2{
+				c.NewChain.Selector: *c.MCMSDeploymentConfig,
+			})
+		}, newAddresses, e.ExistingAddresses)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to run DeployMCMSWithTimelockV2 on chain with selector %d: %w", c.NewChain.Selector, err)
+		}
+	}
+
+	// Deploy chain contracts to the new chain
+	err = runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+		return DeployChainContractsChangeset(e, c.deploymentConfigForNewChain())
+	}, newAddresses, e.ExistingAddresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run DeployChainContractsChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
+
+	// Set RMN remote config & set RMN on proxy on the new chain (if config provided)
+	if c.NewChain.RMNRemoteConfig != nil {
+		_, err = SetRMNRemoteConfigChangeset(e, c.rmnRemoteConfigForNewChain())
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to run SetRMNRemoteConfigChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+		}
+	}
+	// Set the RMN remote on the RMN proxy, using MCMS if RMN proxy is owned by Timelock
+	// RMN proxy will already exist on chains that supported CCIPv1.5.0, in which case RMN proxy will be owned by Timelock
+	state, err := changeset.LoadOnchainState(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	owner, err := state.Chains[c.NewChain.Selector].RMNProxy.Owner(&bind.CallOpts{
+		Context: e.GetContext(),
+	})
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to get owner of RMN proxy on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
+	var mcmsConfig *changeset.MCMSConfig
+	if owner == state.Chains[c.NewChain.Selector].Timelock.Address() {
+		mcmsConfig = c.MCMSConfig
+	}
+	_, err = SetRMNRemoteOnRMNProxyChangeset(e, SetRMNRemoteOnRMNProxyConfig{
+		ChainSelectors: []uint64{c.NewChain.Selector},
+		MCMSConfig:     mcmsConfig,
+	})
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run SetRMNRemoteOnRMNProxyChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
```

Comment:
the proposal append is missing here

Reply:
https://github.com/smartcontractkit/chainlink/pull/16705/commits/b3969487f7890ef00394b4310df85b37b227836a